### ISSUE TITLE
Review misleading error messages

### ISSUE THREAD

**Issue by tripu** [w3c/specberus#536]
In particular, those that are occurring lately because of the switch to HTTPS; we've had a few recently. Users don't notice that the error is a mere &ldquo;s&rdquo; missing (understandably), because the error messages are generic, and some times don't point to the corresponding publication rule, where the boilerplate is.

**Comment by plehegar** [comment]
A complementary way to fix error messages is to move them out of the software and put them in GitHub or a wiki. Ie point people to a page describing the error and, as always, allow folks to contribute to those pages.
Today, going from
     "specberus": {
        "status": "failure",
        "errors": [
          {
            "key": "this-latest-shortname",
            "type": {
              "name": "headers.dl",
              "section": "front-matter",
              "rule": "docIDFormat"
            }
          }
        ]
      },

to

https://github.com/w3c/specberus/blob/master/lib/l10n-en_GB.js#L45

isn't user friendly.

**Comment by tripu** [comment]
I agree that the errors we have now in JSON aren't good. But I personally prefer descriptive error messages to be contained in the software itself.

I started working on this in w3c/echidna#431 already.

Errors will look more or less like this:

> * 👎 [Error code “wrong-date”](https://www.w3.org/pubrules/doc/rules/?profile=WD#dateTitleH2)
> * 👎 [HTTP resource not retrievable. The HTTP status from the remote server was: 404.](https://www.w3.org/pubrules/doc/rules/?profile=WD#valideHTML)

These changes are reformatting the errors in Echidna; but the same transformation I hope we can apply directly in Specberus.

**Comment by plehegar** [comment]
again, we should be able to point to a page for further information, so that we can easiuly extend the documentation beyond simple sentence.

**Comment by tripu** [comment]
@plehegar, the rules on [`w3.org/pubrules/doc`](https://www.w3.org/pubrules/doc/) are sometimes several sentences long, and include examples and suggested boilerplate. [eg](https://www.w3.org/pubrules/doc/rules/?profile=WD#patPolReq), [eg](https://www.w3.org/pubrules/doc/rules/?profile=WD#docIDOrder). Do you think that's still not enough?
